Source/WebService/Blackjack.WebService/ClientApp/src/app/components/play-game-page/play-game.component.ts

```diff
@@ -200,28 +200,33 @@ export class PlayGameComponent implements OnInit {
     this.isGameButtonHidden = true;
   }
 
-  /*TO DO: implement update user endpoint*/
+  /**
+   * Resolves any situation in which the player wins a hand of blackjack
+   */
   public winOutcome(): void {
     this.user.balance += (this.wager * 2);
     this.wager = 0;
+    this._userApi.updateUser(this.user);
```

Comment:
this needs to be awaited

```javascript
await this._userApi.updateUser(this.user);
```

Which means winOutcome needs to be an async function.

Which would mean winOutcome needs to be await

```javascript
await this.winOutcome();
```

Which means any function using winOutcome need to be async

Which means...

You see the pattern.

Reply:
I'm changing it but I am curious why my program seemed to work without that?

Reply:
If you don't await the request your code continues to run and does not wait updateUser to complete, just like it does in C#

```typescript
public someFunc(): void {
    const myResponse: any = this.myApi.myAsyncOperation();
    if (myResponse.someProperty === 'someValue') {
        alert("the thing equaled the other thing!");
    } else {
        alert('the thing did not equal the other thing!');
    }
}
```

In this scenario, you did not await `myAsyncOperation` so the http request fires, has not finished, and we immediately move on to the if statement.

We try to access `myResponse.someProperty` well myResponse has no value yet so it will be the JS value `undefined`. So really what we've tried to do is go `undefined.someProperty` there is no prop called, `someProperty` on `undefined` so you'll get a runtime error of, "cannot access property, "someProperty" on, undefined" which is basically the JS equivalent of C#'s NullReferenceException

When you properly await it like this

```typescript
public async someFunc(): Promise<void> {
    const myResponse: any = await this.myApi.myAsyncOperation();
    if (myResponse.someProperty === 'someValue') {
        alert("the thing equaled the other thing!");
    } else {
        alert('the thing did not equal the other thing!');
    }
}
```

Now the if statement code will NOT run until the entirety of `this.myApi.myAsyncOperation` has completed guaranteeing that `myResponse` has a value for when it is referenced below.

All of this is no different than async/await in C# and why we need to use the await keyword for async methods in C#.